src/Controller/ModulesController.php

```diff
@@ -334,4 +365,104 @@ public function upload()
+        $status = $this->request->getData('status');
+        if (!empty($ids)) { // export selected (filter by id)
+            $ids = explode(',', $ids);
+            foreach ($ids as $id) {
+                $data = [
+                    'id' => $id,
+                    'status' => $status,
+                ];
+                $this->apiClient->saveObject($this->objectType, $data);
+            }
+        }
+
+        return $this->redirect(['_name' => 'modules:list', 'object_type' => $this->objectType]);
+    }
+
+    /**
+     * Export data in csv format
+     *
+     * @return void
+     */
+    public function export() : void
+    {
+        $this->request->allowMethod(['post']);
+        $fields = ['id', 'title', 'description', 'uname', 'status', 'body', 'lang', 'extra'];
+        $delimiter = ',';
+        $ids = $this->request->getData('ids');
+        $csv = implode($delimiter, $fields) . "\n";
+        if (!empty($ids)) { // export selected (filter by id)
+            $response = $this->apiClient->getObjects($this->objectType, ['filter' => ['id' => $ids]]);
+            $csv .= $this->csvRows($response, $fields, $delimiter);
+            $csv = rtrim($csv, "\n"); // remove last newline
+            $this->set('csv', $csv);
+        } else { // export all
+            $response = $this->apiClient->getObjects($this->objectType, [
+                'page' => 1,
+                'page_size' => 100,
+            ]);
+            while (!empty($response['links']['next'])) {
+                $csv .= $this->csvRows($response, $fields, $delimiter);
+                $response = $this->apiClient->get($response['links']['next']);
+            }
+            $csv .= $this->csvRows($response, $fields, $delimiter);
+            $csv = rtrim($csv, "\n"); // remove last newline
+        }
+
+        // Output csv file
+        header("Content-type: text/csv");
+        header("Content-Disposition: attachment; filename=" . $this->objectType . "_" . date('Ymd-His') . ".csv");
+        header('Content-Transfer-Encoding: binary');
+        header('Expires: 0');
+        header('Cache-Control: must-revalidate, post-check=0, pre-check=0');
+        header("Pragma: no-cache");
+        print $csv;
+        exit;
+    }
+
+    /**
+     * Csv rows per response data
+     *
+     * @param array $response the Response
+     * @param array $fields the Fields to export for object
+     * @param string $delimiter the Delimiter char
+     * @return string The csv data
+     */
+    private function csvRows($response = [], $fields = [], $delimiter = ',') : string
+    {
+        if (empty($response) || empty($response['data'])) {
+            return '';
+        }
+        $csv = '';
+        foreach ($response['data'] as $index => $data) {
+            $cells = [];
+            foreach ($fields as $field) {
+                if (!empty($data[$field])) {
+                    $cells[] = '"' . preg_replace('/"/', '""', $data[$field]) . '"';
```

Comment:
Why not use `http://php.net/manual/es/function.fputcsv.php` to a temp file?
Ordered fields array may be extracted with ```array_intersect_key($data['attributes'], array_flip($fields))```